code-experiments/src/c_linear.c

```diff
@@ -217,9 +217,17 @@ static coco_problem_t *c_linear_shuffle(coco_problem_t *problem_c,
   
   /* Pick up a random constraint number other than 1.
    * formula for U[M,N]: 
-   * M + rand() / (RAND_MAX / (N - M + 1) + 1)
+   * random = M + (rand() / (RAND_MAX + 1.0)) * (N - M + 1)
```

Comment:
AFAICS the formula is for integers and does not produce the correct result if we aim at a continuous interval. You can easily verify this by setting `N - M == 1` that the size of the interval will be close to 2 instead of 1.